### PR TITLE
Feature/hierarchy

### PR DESCRIPTION
**Pull request by tomivirkki**
Connects to #987

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-grid/1057)

### CODE REVIEW

File: vaadin-grid-hierarchy-toggle.html [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+<script>
+  {
+    /**
+     * `<vaadin-grid-hierarchy-toggle>` is a helper element for the `<vaadin-grid>`
+     * that provides toggle and level display functionality for the item hierarchy.
+     *
+     * #### Example:
+     * ```html
+     * <vaadin-grid>
+     *   <vaadin-grid-hierarchy-toggle frozen></vaadin-grid-hierarchy-toggle>
```
Comment (jouni) [inline vaadin-grid-hierarchy-toggle.html]: This example is outdated. The hierarchy toggle should be wrapped inside a regular column, right?

File: vaadin-grid-hierarchy-toggle.html [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+     *
+     *   <vaadin-grid-column><template>[[index]]<template></vaadin-grid-column>
```
Comment (limonte) [inline vaadin-grid-hierarchy-toggle.html]: < **/** template>

File: vaadin-grid-hierarchy-toggle.html (hunk 1 above)
Comment (tomivirkki) [reply vaadin-grid-hierarchy-toggle.html]: Indeed, fixed

**Comment by platosha** [comment]
Reviewed 2 of 11 files at r5, 3 of 13 files at r6, 2 of 10 files at r7, 1 of 6 files at r8, 14 of 17 files at r9, 1 of 2 files at r10, 2 of 2 files at r11.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/1057)*

**Comment by platosha** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/1057#-:-KxNQ6TOD6iB2qa0Te8Y:bnfp4nl)*

**Pull request merged**